src/_gettsim/policy_environment_postprocessor.py

```diff
@@ -630,30 +423,27 @@ def aggregate_by_group_func(source_col, group_id):
         else:
             raise ValueError(f"Aggr {agg_specs['aggr']} is not implemented.")
 
-    return aggregate_by_group_func
+    return DerivedFunction(
+        aggregate_by_group_func,
+        function_name=agg_col,
+    )
```

Comment:
```suggestion
        derived_from=derived_from,
    )
```

In if/else block above, incorporate:

```py
    if agg_specs["aggr"] == "count":
        derived_from = group_id
    else:
        derived_from = (source_col, group_id)
```